dump/src/reader/v1/mod.rs

```diff
@@ -1,173 +1,265 @@
     }
-}
 
-impl IndexReader for &V1IndexReader {
-    type Document = serde_json::Map<String, serde_json::Value>;
-    type Settings = settings::Settings;
+    pub fn version(&self) -> Version {
+        Version::V1
+    }
 
-    fn name(&self) -> &str {
-        todo!()
+    pub fn date(&self) -> Option<OffsetDateTime> {
+        None
     }
 
-    fn documents(&self) -> Result<Box<dyn Iterator<Item = Result<Self::Document>>>> {
-        todo!()
+    pub fn indexes(&self) -> Result<impl Iterator<Item = Result<V1IndexReader>> + '_> {
+        Ok(self.indexes.iter().map(|index| index.open()))
     }
+}
 
-    fn settings(&self) -> Result<Self::Settings> {
-        todo!()
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Index {
+    pub name: String,
+    pub uid: String,
+    #[serde(with = "time::serde::rfc3339")]
+    created_at: OffsetDateTime,
+    #[serde(with = "time::serde::rfc3339")]
+    updated_at: OffsetDateTime,
+    pub primary_key: Option<String>,
+}
+
+#[derive(Clone)]
+pub struct IndexMetadataV1 {
+    pub name: String,
+    pub metadata: crate::IndexMetadata,
+}
+
+impl From<Index> for IndexMetadataV1 {
+    fn from(index: Index) -> Self {
+        IndexMetadataV1 {
+            name: index.name,
+            metadata: crate::IndexMetadata {
+                uid: index.uid,
+                primary_key: index.primary_key,
+                created_at: index.created_at,
+                updated_at: index.updated_at,
+            },
+        }
     }
 }
 
-impl DumpReader for V1Reader {
-    type Document = serde_json::Map<String, serde_json::Value>;
-    type Settings = settings::Settings;
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Metadata {
+    pub indexes: Vec<Index>,
+    pub db_version: String,
+    pub dump_version: crate::Version,
+}
 
-    type Task = update::UpdateStatus;
-    type UpdateFile = Infallible;
+#[cfg(test)]
+pub(crate) mod test {
+    use std::fs::File;
+    use std::io::BufReader;
 
-    type Key = Infallible;
+    use flate2::bufread::GzDecoder;
+    use meili_snap::insta;
+    use tempfile::TempDir;
 
-    fn date(&self) -> Option<OffsetDateTime> {
-        None
-    }
+    use super::*;
 
-    fn version(&self) -> Version {
-        Version::V1
-    }
+    #[test]
+    fn read_dump_v1() {
+        let dump = File::open("tests/assets/v1.dump").unwrap();
+        let dir = TempDir::new().unwrap();
+        let mut dump = BufReader::new(dump);
+        let gz = GzDecoder::new(&mut dump);
+        let mut archive = tar::Archive::new(gz);
+        archive.unpack(dir.path()).unwrap();
+
+        let dump = V1Reader::open(dir).unwrap();
+
+        // top level infos
+        assert_eq!(dump.date(), None);
+
+        // indexes
+        let mut indexes = dump.indexes().unwrap().collect::<Result<Vec<_>>>().unwrap();
+        // the index are not ordered in any way by default
+        indexes.sort_by_key(|index| index.metadata().uid.to_string());
```

Comment:
```suggestion
```

Thus we don't need these lines

Reply:
Lines are removed, thanks for spotting that!